roles/grafana_agent/tasks/main.yml

```diff
@@ -0,0 +1,68 @@
+---
+- name: Include secrets
+  include_vars: "{{ secrets_path | mandatory }}/mimir_password.yml"
+  no_log: true
+  tags:
+    - always
+
+- name: "Import Grafana GPG key"
+  become: true
+  ansible.builtin.get_url:
+    url: "{{ grafana_apt_repo_key_url }}"
+    dest: /etc/apt/keyrings/grafana.gpg
+    mode: '0644'
+    force: true
+  when: ansible_pkg_mgr == "apt"
+
+- name: Ensure downloaded file for key is a binary keyring
+  shell: "cat /etc/apt/keyrings/grafana.gpg | gpg --dearmor | sudo tee /etc/apt/keyrings/grafana.gpg > /dev/null"
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to APT sources list"
+  become: true
+  ansible.builtin.apt_repository:
+    repo: "deb [signed-by=/etc/apt/keyrings/grafana.gpg] {{ grafana_apt_repo_url }} stable main"
+    state: present
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to yum/dnf systems"
+  become: true
+  ansible.builtin.yum_repository:
+    baseurl: "{{ grafana_rpm_repo_url }}"
+    name: "grafana"
+    description: "grafana"
+    gpgcheck: true
+    gpgkey: "{{ grafana_rpm_repo_key_url }}"
+    state: present
+  when: ansible_os_family == "RedHat"
+
+- name: "Install grafana-agent"
+  become: true
+  ansible.builtin.package:
+    name: "grafana-agent"
+    state: "present"
+
+- name: "Making sure prometheus service is not running since its using the same port as grafana-agent 9090"
+  become: true
+  ansible.builtin.service:
+    name: "prometheus"
+    state: "stopped"
+    enabled: false
```

Comment:
This is kind of dangerous.  Are we running prometheus anywhere?

Reply:
And won't the next task just fail if we are anyway?

Reply:
Yes we are currently running prometheus only on the sepia-grafana server, I think the change I pushed right now is a better solution, the playbook will only check if prometheus listens and it won't continue until it's resolved